hal/halx86/CMakeLists.txt

```diff
@@ -15,30 +15,12 @@ function(add_hal _halname)
     spec2def(${_halname}.dll ../hal.spec ADD_IMPORTLIB)
 
     # Create the actual target
-    if(NOT MSVC)
-        foreach(_component ${_haldata_COMPONENTS})
-            list(APPEND _haldata_SOURCES "$<TARGET_OBJECTS:lib_hal_${_component}>")
-        endforeach()
-        add_library(${_halname} MODULE
-            ${_haldata_SOURCES}
-            ${CMAKE_CURRENT_BINARY_DIR}/hal.def)
-    else()
-        foreach(_component ${_haldata_COMPONENTS})
-            list(APPEND _haldata_LIBS "lib_hal_${_component}")
-        endforeach()
-        add_library(${_halname} MODULE
-            ${_haldata_SOURCES}
-            ${CMAKE_CURRENT_BINARY_DIR}/hal.def)
-        target_link_libraries(${_halname} ${_haldata_LIBS})
-    endif()
-
-    if(${_halname} STREQUAL "halpc98")
-        target_compile_definitions(lib_hal_pc98 PRIVATE SARCH_PC98)
-    endif()
-
-    if(${_halname} STREQUAL "halxbox")
-        target_compile_definitions(lib_hal_xbox PRIVATE SARCH_XBOX)
-    endif()
+    foreach(_component ${_haldata_COMPONENTS})
```

Comment:
And what happens now to the usage of the specific compile-time definitions necessary for these different versions of the HAL?

Reply:
EDIT: Ok it may appear that putting these in their corresponding cmake files do the trick indeed....

Reply:
it sure does. I don't get why they where in this function in the first place.